src/main.rs

```diff
@@ -1,5 +1,12 @@
 use std::collections::HashMap;
 
+use crate::CommsError::{
+    ConnectionClosed, ConnectionExists, ConnectionNotFound, ServerLimitReached, UnexpectedHandshake,
+};
+use crate::Connection::{Closed, Open};
+use crate::MessageType::{GetCount, Handshake, Post};
+use crate::Response::{HandshakeReceived, PostReceived};
```

Comment:
Zbędne importy. Jeśli chodziło o używanie nazw bez ich namespace'ów to i tak nie korzystasz z tego konsekwentnie w kodzie.